db/notifications.go

```diff
@@ -167,7 +171,7 @@ func GetMatchingNotificationSilences(ctx context.Context, resources models.Notif
 	query = query.Where(orClauses)
 
 	var silences []models.NotificationSilence
-	err := query.Where(`"from" <= NOW()`).Where("until >= NOW()").Where("deleted_at IS NULL").Find(&silences).Error
+	err := query.Where(`"from" <= NOW()`).Where("error IS NULL").Where("until >= NOW()").Where("deleted_at IS NULL").Find(&silences).Error
```

Comment:
We might need to null out error, when the filter is updated as otherwise the silence is ignored forever